woocommerce/woocommerce-android
# Issue/2322 update product type

Fixes #2322 by adding the option to update the product type.

#### Changes
There are a few changes in this PR that were necessary in order to build the project.
- This [FluxC PR](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/1659) was added to this PR in order to support updating product type. There were some changes made to FluxC to support encrypted logging for the WordPress apps. So I needed to add the changes in the 1aee952 in order to make sure the app was building.
- The product types should be displayed in a bottom sheet and in order to pass the results back to the `ProductDetailFragment`, I included the changes @0nko made for variations, by updating the navigation component and basically reusing his work with variations, which is currently in a separate branch. This allowed me to return a result from a dialog to the `ProductDetailFragment`.

#### Screenshots
<img src="https://user-images.githubusercontent.com/22608780/89755001-81f22d00-dafb-11ea-8093-c5232a02708f.png" width="300"/>. <img src="https://user-images.githubusercontent.com/22608780/89755003-83bbf080-dafb-11ea-92da-bdb31745f221.png" width="300"/>. <img src="https://user-images.githubusercontent.com/22608780/89755007-86b6e100-dafb-11ea-8f1c-a809c3806204.png" width="300"/>. <img src="https://user-images.githubusercontent.com/22608780/89755012-89193b00-dafb-11ea-8a97-1f2a08032dfb.png" width="300"/>

#### Testing
- Click on a **PHYSICAL** product from the Products tab.
  - Click on the Product Type section. 
  - Change the product type to anything else and verify that a confirmation dialog is displayed. 
    - Click on `Cancel` and notice that the dialog is dismissed.
  - Change the product type to anything else and verify that a confirmation dialog is displayed. 
    - Click on `Change` and notice that the product type is changed in the Product Detail screen.
    - Notice the "UPDATE" menu button is displayed.
    - Click on the "Update" menu button and verify that the product type is updated successfully.
- Follow the above steps for a **Virtual**, **Grouped** and **External** products.

~**This PR is in draft till this [FluxC PR](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/1659) can be reviewed and merged**~

Update release notes:
